src/data_storage/filter.ts

```diff
@@ -0,0 +1,55 @@
+// Filter is the base class for data storage object filters. Filter will allow
+// you to build a class for an individual data type so that you can write complex
+// queries.
+import Result from "../result";
+import PostgresStorage from "./postgresStorage";
+
+export default abstract class Filter {
+    private readonly _tableName: string
+    private _rawQuery: string[] = []
+    private _values: any[] = []
+
+
+    protected constructor(tableName: string) {
+        this._tableName = tableName
+
+        this._rawQuery.push(`SELECT * FROM ${this._tableName}`)
+    }
+
+    where() {
+        this._rawQuery.push("WHERE")
+        return this;
+    }
+
+    and() {
+        this._rawQuery.push("AND")
+        return this
+    }
+
+    query(fieldName: string, operator: string, value: any) {
+        switch(operator) {
+            case "eq": {
+               this._rawQuery.push(`${fieldName} = $${this._values.length+1}`);
```

Comment:
What does the $$ do here? Also does adding value to _values not need to happen first?

Reply:
D'oh, I could add the value to values first you're right, then no need for the `+1`. The double ampersands are a mix of the notation needed for the SQL Query config for postgres `$1` and string interpolation in typescript `${}`